src/ScottPlot5/ScottPlot5/Plottables/Legend.cs

```diff
@@ -0,0 +1,188 @@
+            }
+        }
+
+        private void RenderLegendSymbol(SKSurface surface, LegendItem item, SKRect rect)
+        {
+            using SKPaint paint = new();
+
+            if (item.Line.HasValue)
+            {
+                paint.SetStroke(item.Line.Value);
+
+                surface.Canvas.DrawLine(new(rect.Left, rect.MidY), new(rect.Right, rect.MidY), paint);
+            }
+
+            if (item.Marker.HasValue)
+            {
+                paint.Style = SKPaintStyle.Fill;
+                paint.Color = item.Marker.Value.Color.ToSKColor();
+
+                if (item.Marker.Value.Shape == MarkerShape.Circle)
+                {
+                    surface.Canvas.DrawCircle(new(rect.MidX, rect.MidY), item.Marker.Value.Size / 2, paint);
+                }
+                else
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            if (item.Fill.HasValue)
+            {
+                paint.SetFill(item.Fill.Value);
+
+                surface.Canvas.DrawRect(rect, paint);
+            }
+        }
+
+        public PixelSize Measure(LegendItem item, bool includeChildren = true)
+        {
+            if (ShouldHide(item))
+            {
+                return PixelSize.Zero;
+            }
+
+            using SKPaint paint = new(Font);
+
+            PixelSize labelRect = Drawing.MeasureString(item.Label ?? "", paint);
+
+            float width = HorizontalPadding + SymbolWidth + HorizontalPadding + labelRect.Width + HorizontalPadding;
+            float height = string.IsNullOrEmpty(item.Label) ? VerticalPadding / 2 : 2 * VerticalPadding + labelRect.Height;
+            var childrenArray = item.Children.ToArray(); // Avoids enumerating the children enumerable multiple times
+
+            if (childrenArray.Length > 0)
+            {
+                width += HorizontalPadding;
+            }
+
+            if (includeChildren)
+            {
+                for (int i = 0; i < childrenArray.Length; i++)
+                {
+                    var childSize = Measure(childrenArray[i]);
+                    width = Math.Max(width, HorizontalPadding + childSize.Width);
+                    height += childSize.Height;
+                }
+            }
+
+            return new(width, height);
+        }
+
+        // I'm a Javascript developer, this was bound to happen someday
+        public PixelSize Measure() =>
+            LegendItems
+                .Select((LegendItem item) => Measure(item))
+                .Aggregate(
+                    new PixelSize(Border.Width * 2, Border.Width * 2),
+                    (a, b) =>
+                        new(
+                            Math.Max(a.Width, b.Width),
+                            a.Height + b.Height)
+                        );
+
+        private Pixel GetTopLeftCorner(PixelSize size)
+        {
+            const int margin = 5;
+
+            return Position switch
```

Comment:
Maybe this logic could be moved inside `Corner` because it seems like a common task